includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -263,6 +263,20 @@ public function create_intention(
 		$request['payment_method_types'] = $payment_methods;
 		$request['capture_method']       = $capture_method;
 
+		if ( $this->is_fraud_prevention_enabled_for_store() ) {
+			$request['metadata']['fraud_prevention_data_available'] = true;
+			$order               = wc_get_order( $order_id );
+			$billing_email_parts = explode( '@', $order->get_billing_email() );
+			$email_domain        = array_pop( $billing_email_parts );
+
+			$request['metadata']['fraud_prevention_data'] = [
```

Comment:
Similar to https://github.com/Automattic/woocommerce-payments/pull/3979/files. we should introduce a `Buyer_Fingerprinting_Service` which will take care of the metadata generation for the order.